File: android-lib/src/main/java/com/michel/intrastructure/mvc/BaseActivity.java
package com.michel.intrastructure.mvc;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

/**
 * MVC-Base
 * Created by Michel on 2017/4/26.
 */

public abstract class BaseActivity extends AppCompatActivity{

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResId());

        initVariables();

        initViews(savedInstanceState);

        initData();
    }

    protected abstract int getLayoutResId();

    protected abstract void initVariables();

    protected abstract void initViews(Bundle savedInstanceState);

    protected abstract void initData();
}

File: androidlib/src/main/java/com/michel/intrastructure/activity/BaseActivity.kt
package com.michel.intrastructure.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.michel.intrastructure.utils.*

/**
 * Activity 基类
 * Created by Michel on 2017/3/29.
 */

abstract class BaseActivity : AppCompatActivity(){

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(layoutResId)

        initVariables()

        initViews(savedInstanceState!!)

        initData()

        LogD(javaClass.simpleName)
    }



    protected abstract val layoutResId: Int

    protected abstract fun initVariables()

    protected abstract fun initViews(savedInstanceState: Bundle)

    protected abstract fun initData()
}
